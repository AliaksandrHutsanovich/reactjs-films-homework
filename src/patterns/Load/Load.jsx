/* eslint react/self-closing-comp: 0 */
import React, { memo } from 'react';
import { Text } from '../../components';

import styles from './Load.scss';

const Load = () => (
  <div className={styles.wrapper}>
    <ul className={styles.wrapper__list}>
      <li><div className={styles.wrapper__list_item}></div></li>
      <li><div className={styles.wrapper__list_item}></div></li>
      <li><div className={styles.wrapper__list_item}></div></li>
    </ul>
    <Text type="normal">LOADING</Text>
  </div>
);

export default memo(Load);
