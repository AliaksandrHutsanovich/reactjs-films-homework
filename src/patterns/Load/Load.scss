@font-face{
  font-family: 'HelveticaNeue';
  src: url('../../assets/fonts/HelveticaNeue.ttf') format('truetype');
}

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  &__list {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 50px;
    list-style-type: none;
    padding: 0;
    height: 10px;

    &>:nth-child(n) {
      position: relative;
    }

    &>:first-child > div {
      animation: li1 2s infinite linear;
    }

    &>:nth-child(2) > div {
      animation: li2 2s infinite linear;
    }

    &>:nth-child(3) > div {
      animation: li3 2s infinite linear;
    }
  }

  &__list_item {
    background-color: rgb(10, 174, 228);
    border-radius: 4px;
    width: 6px;
    height: 6px;
    position: absolute;
    top: -3px;
    left: -3px;
  }

  &__text {
    text-align: center;
    color: white;
    font: normal 15px HelveticaNeue;
  }
}

@keyframes rot {
  0% {
    transform: rotate(0deg);
  }
  25% {
    transform: rotate(90deg);
  }
  50% {
    transform: rotate(180deg);
  }
  75% {
    transform: rotate(270deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes li1 {
  0% {
    transform: scale(1);
  }
  14.3% {
    transform: scale(2.3);
  }
  28.6% {
    transform: scale(1);
  }
  42.9% {
    transform: scale(1);
  }
  57.2% {
    transform: scale(2.3);
  }
  71.5% {
    transform: scale(1);
  }
  85.8% {
    transform: scale(1);
  }
  100% {
    transform: scale(1);
  }
}

@keyframes li2 {
  0% {
    transform: scale(1);
  }
  14.3% {
    transform: scale(1);
  }
  28.6% {
    transform: scale(2.3);
  }
  42.9% {
    transform: scale(1);
  }
  57.2% {
    transform: scale(1);
  }
  71.5% {
    transform: scale(2.3);
  }
  85.8% {
    transform: scale(1);
  }
  100% {
    transform: scale(1);
  }
}

@keyframes li3 {
  0% {
    transform: scale(1);
  }
  14.3% {
    transform: scale(1);
  }
  28.6% {
    transform: scale(1);
  }
  42.9% {
    transform: scale(2.3);
  }
  57.2% {
    transform: scale(1);
  }
  71.5% {
    transform: scale(1);
  }
  85.8% {
    transform: scale(2.3);
  }
  100% {
    transform: scale(1);
  }
}
