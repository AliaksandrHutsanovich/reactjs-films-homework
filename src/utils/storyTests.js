const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
