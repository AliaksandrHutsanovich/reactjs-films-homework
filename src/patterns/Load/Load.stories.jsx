import React from 'react';
import Load from './Load';

export default {
  component: Load,
  title: 'patterns/Load',
};

export const loadExample = () => (
  <Load />
);
