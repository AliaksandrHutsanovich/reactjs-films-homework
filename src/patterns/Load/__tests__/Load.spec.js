import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../Load.stories';

describe('<Load />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
