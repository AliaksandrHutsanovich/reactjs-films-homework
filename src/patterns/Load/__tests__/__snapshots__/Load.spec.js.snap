// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Load /> loadExample should be rendered properly 1`] = `
<div
  className="wrapper"
>
  <ul
    className="wrapper__list"
  >
    <li>
      <div
        className="wrapper__list_item"
      />
    </li>
    <li>
      <div
        className="wrapper__list_item"
      />
    </li>
    <li>
      <div
        className="wrapper__list_item"
      />
    </li>
  </ul>
  <UNDEFINED
    type="normal"
  >
    LOADING
  </UNDEFINED>
</div>
`;
